### PR TITLE
feat: add GuildPreview

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

Add GuildPreview, which allows viewing a preview Guild for any discoverable guild.
## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint` or `pre-commit run --all-files`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/client.py [hunk 1]
```diff
@@ -1568,6 +1569,40 @@ async def fetch_guild(self, guild_id: int, /) -> Guild:
...
+
+        Parameters
+        ----------
+        guild_id: :class:`int`
+            The ID of the guild to to retrieve a preview object.
+
+        Raises
+        ------
+        NotFound
+            Retrieving the guild preview failed.
```
Comment (EQUENOS) [inline disnake/client.py:1596]: What about `Forbidden` and `HTTPException`?

File: disnake/guild_preview.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+if TYPE_CHECKING:
+    from .state import ConnectionState
+    from .types.guild import GuildPreview as GuildPreviewPayload
+
+__all__ = ("GuildPreview",)
+
+
+class GuildPreview:
+    """Represents a :class:`.Guild` preview object..
```
Comment (EQUENOS) [inline disnake/guild_preview.py:40]: Too many dots

File: disnake/guild_preview.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        else:
+            self.stickers: Tuple[GuildSticker, ...] = ()
+
+    def __repr__(self) -> str:
+        attrs = (
+            ("id", self.id),
+            ("name", self.name),
+        )
+        inner = " ".join("%s=%r" % t for t in attrs)
+        return f"<GuildPreview {inner}>"
```
Comment (EQUENOS) [inline disnake/guild_preview.py:118]: `return f"<GuildPreview id={self.id!r} name={self.name!r}>"`

**Review by EQUENOS — COMMENTED** [review]
Review

File: disnake/client.py (hunk 1 above)
Comment (onerandomusername) [reply disnake/client.py:1596]: HTTPException missing is an oversight for a malformed request, but if you do not have access to a guild's preview, its a 404.

File: disnake/guild_preview.py (hunk 2 above)
Comment (onerandomusername) [reply disnake/guild_preview.py]: https://github.com/DisnakeDev/disnake/pull/359/commits/60d82aef166e59fff368e425c95cdc53201b4609

File: disnake/guild_preview.py (hunk 3 above)
Comment (onerandomusername) [reply disnake/guild_preview.py]: https://github.com/DisnakeDev/disnake/pull/359/commits/52022cd69a1958deeffcbf7c457099a74bb7a309

**Review by EQUENOS — APPROVED** [review]
Lgtm!

**Pull request merged**